commit 6b833549a5739a30c623198e9029765e2b1fa3ff (refs/changes/30/1730/3)
Author: Joshua M. Clulow <jmc@joyent.com>
Date:   2017-04-05T06:01:22+00:00 (2 years, 6 months ago)
    
    OS-5918 allow admin_ip to be set to "none" in config
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>
    Approved by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/overlay/generic/lib/svc/method/net-physical b/overlay/generic/lib/svc/method/net-physical
index a4dcc8b4..7fbae783 100644
--- a/overlay/generic/lib/svc/method/net-physical
+++ b/overlay/generic/lib/svc/method/net-physical
@@ -522,8 +522,9 @@ if smf_is_globalzone; then
         admin_ip6=${BOOT_admin_ip6}
     fi
 
-    if [[ -n ${admin_ip} ]] && [[ -n ${admin_netmask} ]];
-    then
+    if [[ $admin_ip == 'none' ]]; then
+        echo 'INFO: not configuring IP on admin interface (admin_ip=none)'
+    elif [[ -n $admin_ip ]] && [[ -n $admin_netmask ]]; then
         /sbin/ifconfig ${SYSINFO_NIC_admin} inet ${admin_ip} \
             netmask ${admin_netmask} up
         ADMIN_NIC_UP=true
@@ -817,6 +818,13 @@ if [[ -n "${ADMIN_NIC_MISCONFIGURED}" ]]; then
     exit ${SMF_EXIT_ERR_CONFIG}
 fi
 
+if [[ $admin_ip == 'none' ]]; then
+    #
+    # We're done, even if there are not any usable IP addresses.
+    #
+    exit $SMF_EXIT_OK
+fi
+
 # Any non-loopback IPv4 interfaces with usable addresses up?
 if [[ -n "`/sbin/ifconfig -a4u`" ]]; then
     /sbin/ifconfig -a4u | while read intf addr rest; do
